test/utils/runners.go

```diff
@@ -1336,21 +1336,41 @@ func MakePodSpec() v1.PodSpec {
 	}
 }
 
-func makeCreatePod(client clientset.Interface, namespace string, podTemplate *v1.Pod) error {
-	if err := CreatePodWithRetries(client, namespace, podTemplate); err != nil {
+func makeCreatePod(client clientset.Interface, namespace string, podTemplate *v1.Pod, counter int) error {
+	var pod *v1.Pod
+	if err := InjectCounter(counter, podTemplate, &pod); err != nil {
+		return fmt.Errorf("injecting --counter: %v", err)
+	}
+
+	if err := CreatePodWithRetries(client, namespace, pod); err != nil {
 		return fmt.Errorf("error creating pod: %v", err)
 	}
 	return nil
 }
 
+// InjectCounter marshals the object with JSON, replaces the `--counter` string (chosen
+// that way because it is valid inside most fields as suffix which unlikely to occur
+// by chance), and umarshals the modified object.
+func InjectCounter(counter int, in, out interface{}) error {
```

Comment:
This seems too generic and obscure.
Can you make it about v1.Pod specifically?

Reply:
+1

Reply:
/close

Not really needed.